traversal/planner/PlannerEdge.java

```diff
@@ -273,57 +256,34 @@ private void recordCost() {
             costLastRecorded = costNext;
         }
 
-        private void recordResults() {
-            varIsSelected_result = (int) Math.round(varIsSelected.solutionValue());
-            varOrderNumber_result = (int) Math.round(varOrderNumber.solutionValue());
-        }
-
         private void resetInitialValue() {
+            varIsSelected.clearInitial();
+            varOrderNumber.clearInitial();
+            iterate(varOrderAssignment).forEachRemaining(BoolVariable::clearInitial);
             hasInitialValue = false;
-            varIsSelected_init = 0;
-            varOrderNumber_init = 0;
-            Arrays.fill(varOrderAssignment_init, 0);
         }
 
         void setInitialValue(int order) {
-            varOrderNumber_init = order;
-            if (order > 0) {
-                varIsSelected_init = 1;
-                varOrderAssignment_init[order - 1] = 1;
+            assert order > 0;
+            varOrderNumber.setInitial(order);
+            varIsSelected.setInitial(true);
+            for (int i = 0; i < varOrderAssignment.length; i++) {
+                if (i == order - 1) varOrderAssignment[i].setInitial(true);
+                else varOrderAssignment[i].setInitial(false);
```

Comment:
we have to explicitly set the initial `false` values rather than having them be implicitly 0